packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -626,18 +627,26 @@ function createViewModelHook(elm: HTMLElement, vnode: VCustomElement, renderer:
     return vm;
 }
 
-function allocateInSlot(vm: VM, children: VNodes, owner: VM) {
-    const {
-        cmpSlots: { slotAssignments: oldSlotsMapping },
-    } = vm;
-    const cmpSlotsMapping = create(null);
-    vm.cmpSlots = { owner, slotAssignments: cmpSlotsMapping };
+/**
+ * Collects all slots into a SlotSet, traversing through VFragment Nodes
+ *
+ * @param vm
+ * @param children
+ * @param cmpSlots
+ */
+function collectSlots(vm: VM, children: VNodes, cmpSlotsMapping: { [key: string]: VNodes }) {
     for (let i = 0, len = children.length; i < len; i += 1) {
         const vnode = children[i];
         if (isNull(vnode)) {
             continue;
         }
 
+        // Dive further iff the content is wrapped in a VFragment
+        if (isVFragment(vnode)) {
+            collectSlots(vm, vnode.children, cmpSlotsMapping);
```

Comment:
Should we discard the text delimiters since the children are being extracted from the fragment? Currently they are being assigned to the default slot because they do not have a `slot` attribute.